### PR TITLE
dev-cmd/pr-*: allow to run on other taps too

### PR DESCRIPTION
**Pull request by dawidd6**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

I'm thinking aloud about adopting the `homebrew-core` way of publishing bottles into `linuxbrew-core` (to further reduce the diff, but in opposite direction 😄 ), hence this PR. Just a draft now, suggestions and discussions welcomed.

### CODE REVIEW

File: Library/Homebrew/dev-cmd/pr-automerge.rb [hunk 1]
```diff
@@ -59,7 +58,7 @@ def pr_automerge
     end
 
     if args.publish?
-      safe_system "#{HOMEBREW_PREFIX}/bin/brew", "pr-publish", *pr_urls
```
Comment (jonchang) [inline Library/Homebrew/dev-cmd/pr-automerge.rb:62]: We'd need to pass along a `--tap` argument when relevant here.

Comment (dawidd6) [reply Library/Homebrew/dev-cmd/pr-automerge.rb:62]: Done

File: Library/Homebrew/dev-cmd/pr-automerge.rb [hunk 2]
```diff
@@ -59,7 +58,9 @@ def pr_automerge
...
+      publish_args = ["pr-publish"]
+      publish_args << "--tap=#{args.tap}" if args.tap
```
Comment (jonchang) [inline Library/Homebrew/dev-cmd/pr-automerge.rb:62]:
```suggestion
      publish_args << "--tap=#{tap}" if tap
```

File: Library/Homebrew/dev-cmd/pr-publish.rb [hunk 3]
```diff
@@ -22,14 +24,13 @@ def pr_publish_args
...
 
     args.named.uniq.each do |arg|
-      arg = "#{CoreTap.instance.default_remote}/pull/#{arg}" if arg.to_i.positive?
+      arg = "#{tap.default_remote}/pull/#{arg}" if arg.to_i.positive?
       url_match = arg.match HOMEBREW_PULL_OR_COMMIT_URL_REGEX
       _, user, repo, issue = *url_match
       odie "Not a GitHub pull request: #{arg}" unless issue
-      tap = Tap.fetch(user, repo) if repo.match?(HOMEBREW_OFFICIAL_REPO_PREFIXES_REGEX)
       ohai "Dispatching #{tap} pull request ##{issue}"
       GitHub.dispatch_event(user, repo, "Publish ##{issue}", pull_request: issue)
```
Comment (jonchang) [inline Library/Homebrew/dev-cmd/pr-publish.rb:35]: This command takes a `tap` argument but dispatches against `repo`. I feel like this isn't quite correct but I'm not sure what the solution would be.

Comment (dawidd6) [reply Library/Homebrew/dev-cmd/pr-publish.rb:35]: I think it is good as it is now.

File: Library/Homebrew/dev-cmd/pr-publish.rb [hunk 4]
```diff
@@ -22,14 +24,13 @@ def pr_publish_args
...
+    tap = Tap.fetch(Homebrew.args.tap || CoreTap.instance.name)
 
     args.named.uniq.each do |arg|
-      arg = "#{CoreTap.instance.default_remote}/pull/#{arg}" if arg.to_i.positive?
+      arg = "#{tap.default_remote}/pull/#{arg}" if arg.to_i.positive?
       url_match = arg.match HOMEBREW_PULL_OR_COMMIT_URL_REGEX
       _, user, repo, issue = *url_match
       odie "Not a GitHub pull request: #{arg}" unless issue
-      tap = Tap.fetch(user, repo) if repo.match?(HOMEBREW_OFFICIAL_REPO_PREFIXES_REGEX)
       ohai "Dispatching #{tap} pull request ##{issue}"
```
Comment (jonchang) [inline Library/Homebrew/dev-cmd/pr-publish.rb:34]:
```suggestion
      if args.tap.present? && "#{user}/#{repo}".downcase != tap.full_name.downcase
        odie "Pull request URL is for #{user}/#{repo} but --tap=#{tap.full_name}!"
      end

      ohai "Dispatching #{tap} pull request ##{issue}"
```

What do you think?

```console
% brew pr-publish --tap=homebrew/core https://github.com/Homebrew/linuxbrew-core/pull/9001 
Error: Pull request URL is for Homebrew/linuxbrew-core but --tap=Homebrew/homebrew-core!

% brew pr-publish https://github.com/Homebrew/linuxbrew-core/pull/9001 
==> Dispatching homebrew/core pull request #9001

% brew pr-publish --tap=homebrew/core https://github.com/Homebrew/homebrew-core/pull/9001
==> Dispatching homebrew/core pull request #9001
```

**Pull request merged**